File: SmartShop.DataApi/ViewModels/Edit/ProductEditModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace SmartShop.DataApi.ViewModels.Edit
{
    public class ProductEditModel
    {
        public ProductEditModel()
        {
            

            this.ProductImages = new List<ProductImageEditModel>();
            this.ProductSpecs = new List<ProductSpecEditModel>();
            this.ProductPrices = new List<ProductPriceEditModel>();
          
        }
        public int ProductId { get; set; }

        [Required, StringLength(100)]
        public string ProductName { get; set; }

        [Required, StringLength(200)]
        public string ProductDescription { get; set; }

       
        [Required, StringLength(30)]
        public string PriceDeterminingProperty { get; set; }

        public bool? ProductStatus { get; set; }

        [Required]
        public int BrandId { get; set; }

        public int CategoryId { get; set; }
       
        public int SubcategoryId { get; set; }

        public virtual IList<ProductImageEditModel> ProductImages { get; set; }
        public virtual IList<ProductSpecEditModel> ProductSpecs { get; set; }
        public virtual IList<ProductPriceEditModel> ProductPrices { get; set; }
    }
}

File: SmartShop.Client/ClientApp/src/app/models/data/edit/product-edit-model.ts
import { ProductImageEditModel } from "./product-image-edit-model";
import { ProductPriceEditModel } from "./product-price-edit-model";
import { ProductSpecEditModel } from "./product-spec-edit-model";

export interface ProductEditModel {
  productId?: number;
  productName?: string;
  productDescription?: string;
  priceDeterminingProperty?: string;
  productStatus?: boolean;
  brandId?: number;
  categoryId?: number;
  subcategoryId?: number;
  productImages?: ProductImageEditModel[];
  productSpecs?: ProductSpecEditModel[];
  productPrices?: ProductPriceEditModel[];
}
